compiler/test/Dialect/Linalg/fuse-attention.mlir

```diff
@@ -160,3 +160,72 @@ transform.sequence failures(propagate) {
   %0 = transform.structured.match attributes {__root__} in %arg0 : (!pdl.operation) -> !pdl.operation
   %transformed, %loops:2 = transform.structured.fuse_ext %0 {tile_interchange = [0, 1, 4, 3, 2], tile_sizes = [2, 4, 0, 0, 0]}
 }
+
+// -----
+
+// support split-head multi-query-attention, see https://arxiv.org/pdf/1911.02150.pdf
+// batch size = 4
+// sequence length = 1024
+// hidden size = 512
+// head number = 8
+// head dimension = 64
+func.func @multiquery_attention_with_prologue_proj(%arg0: tensor<4x1024x512xf32>, %arg1: tensor<512x512xf32>, %arg2: tensor<512x64xf32>, %arg3: tensor<512x64xf32>) -> tensor<4x8x1024x64xf32> {
+  // CHECK-LABEL: @multiquery_attention_with_prologue_proj
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg_ext.batch_matmul{{.*}}__stop__
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg_ext.batch_matmul{{.*}}__stop__
+  // CHECK: scf.for
+  // CHECK:   scf.for
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg_ext.batch_matmul
+  // CHECK-DAG: tensor.expand_shape
+  // CHECK-DAG: linalg.transpose
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg.broadcast
+  // CHECK-DAG: linalg.transpose
+  // CHECK-DAG: linalg_ext.batch_matmul
+  // CHECK-DAG: linalg_ext.softmax
+  // CHECK-DAG: linalg_ext.batch_matmul
+  // CHECK:     scf.yield
+  // CHECK:   scf.yield
+  %cst = arith.constant 0xFF800000 : f32
```

Comment:
Is this %cst used anywhere?

Reply:
It's used in linalg.fill

Reply:
Ah, okay. Somehow missed that during search.